src/main/java/spoon/reflect/path/CtRole.java

```diff
@@ -99,7 +99,8 @@
 	OPENED_PACKAGE,
 	SERVICE_TYPE,
 	IMPLEMENTATION_TYPE,
-	PROVIDED_SERVICE;
+	PROVIDED_SERVICE,
+	MODULE_MEMBER;
```

Comment:
MODULE_MEMBER is super role of other roles. Move it before the first sub rule, and then use `CtRole(CtRole)` constructor on the sub roles to link them.